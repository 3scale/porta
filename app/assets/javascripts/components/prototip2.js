

var Bubble = Class.create();
Bubble.prototype = {
  initialize: function(el) {
    this.offsetX = -8;
    this.offsetY = 38;

    this.target = el;
    // this.text = text;
    this.makeBubble();
  },
 
  makeBubble: function() {
    var bubble = $("bubble_" + this.target.title);
    // var bubble = new Element('div', { 'class': 'bubble'});
    // var content = new Element('div', { 'class': 'content'});
    // content.innerHTML = this.text;

    // bubble.appendChild(content);
    this.bubble = bubble;
    // new Insertion.Bottom(body, bubble);
    // bubble.hide();
    
    
    
    // set position

    // listen!
    
    this.target.observe('mouseover', (
        function() { 

          var position = this.target.cumulativeOffset();

          this.bubble.style.top = position[1] + this.offsetX + "px";
          this.bubble.style.left = position[0] + this.target.getWidth() + this.offsetY + "px";
          
          this.bubble.show(); 
        }).bind(this));
        
    this.target.observe('mouseout', (
        function() { 
          this.bubble.hide(); 
        }).bind(this));
        
  }
  
  
}

var BubbleHelper = {
  set: function(){
    $$('.define').each(function(l){
      new Bubble(l);      
    });
}}
  //     function(event) { 
  // 
  //       if ((event.element().className == 'submit' )) {
  // 
  //         var form = event.element().up('form');
  //         var formData = form.serialize(true);
  //         var url = form.action;
  //         
  //         this.ajaxPost(formData, url);
  //         event.stop();
  //       }
  //     })
  //   )        
  //   
  // }
