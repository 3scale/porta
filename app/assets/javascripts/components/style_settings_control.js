// Functions dealing with provider side control of site colours. Probably to be deprecated with the
// development of CMS system

jQuery.noConflict();

var settings = function() {
    // var B = ["background_color", "text_color", "link_color", "sidebar_fill_color", "sidebar_border_color"];
    var A = {
        updateColors: function() {
            var E = jQuery("#settings_bg_colour").val();
            var G = jQuery("#settings_text_colour").val();
            var F = jQuery("#settings_link_colour").val();
            var C = jQuery("#settings_menu_bg_colour").val();
						var D = jQuery("#settings_menu_link_colour").val();
						var H = jQuery("#settings_content_bg_colour").val();
						var J = jQuery("#settings_plans_tab_bg_colour").val();
						var K = jQuery("#settings_content_border_colour").val();
						
            jQuery("#previewContent").css({"background-color": E, color: G});

						jQuery('#pBox span').css("color", G);
						jQuery('#pBox  span a').css("color", F);
						jQuery('#pLinks span.active').css("background-color", H);
						jQuery('#pLinks span').css("color", D);
						jQuery('#pBox, #pFooter').css("background-color", H);
						jQuery('#pBox, #pFooter').css("border-color", K);
						jQuery('#previewPlans table th.head').css("background-color", J);
						
            jQuery("#pMenu").css({
                "background-color": C
            });
        },
        synchColors: function(D) {
            var C = settings.colorPicker.linked;
            if (C && C.value && C.value != settings.colorPicker.color) {
                C.value = settings.colorPicker.color;
                jQuery(C).css({
                    "background-color": C.value,
                    color: settings.colorPicker.hsl[2] > 0.5 ? "#000": "#fff"
                });
                settings.updateColors()
            }
        },
        initDesignSettings: function() {

          jQuery("#linkDesign a").click(function() {
              jQuery("#designSettings").show();
							jQuery('#detailSettings').hide();
              jQuery("#linkDetails a").removeClass("active");
              jQuery("#linkDesign a").addClass("active");
              jQuery("#linkDetails").removeClass("active");
              jQuery("#linkDesign").addClass("active");
							
              return false
          });

          jQuery("#linkDetails a").click(function() {
              jQuery("#designSettings").hide();
							jQuery('#detailSettings').show();

              // jQuery("#backgrounds").hide();
              jQuery("#linkDetails a").addClass("active");
              jQuery("#linkDesign a").removeClass("active");
              jQuery("#linkDetails").addClass("active");
              jQuery("#linkDesign").removeClass("active");

              // if (jQuery("#colors").css("display") == "none") {
              //     jQuery("#title_picker").removeClass("active");
              //     jQuery("#title_background").removeClass("inactive")
              // } else {
              //     jQuery("#title_picker").addClass("active");
              //     jQuery("#title_background").addClass("inactive")
              // }
              // if (jQuery("#user_profile_default").val()) {
              //     jQuery("#user_profile_default").val(false)
              // }
              return false
          });

            jQuery("#title_picker a").click(function() {
                jQuery("#colors").toggle();
                jQuery("#backgrounds").hide();
                jQuery("#title_background").removeClass("active");
                if (jQuery("#colors").css("display") == "none") {
                    jQuery("#title_picker").removeClass("active");
                    jQuery("#title_background").removeClass("inactive")
                } else {
                    jQuery("#title_picker").addClass("active");
                    jQuery("#title_background").addClass("inactive")
                }
                if (jQuery("#user_profile_default").val()) {
                    jQuery("#user_profile_default").val(false)
                }
                return false
            });
            jQuery("#title_background a").click(function() {
                jQuery("#backgrounds").toggle();
                jQuery("#colors").hide();
                jQuery("#title_picker").removeClass("active");
                if (jQuery("#backgrounds").css("display") == "none") {
                    jQuery("#title_background").removeClass("active");
                    jQuery("#title_picker").removeClass("inactive")
                } else {
                    jQuery("#title_background").addClass("active");
                    jQuery("#title_picker").addClass("inactive")
                }
                return false
            });
            jQuery("#themes a").click(function() {
                var E = jQuery(this).attr("id");
                settings.setTheme(E.substring(5));
                return false
            });
            jQuery("#backgrounds a").click(function() {
                jQuery("#backgrounds a").removeClass("active");
                jQuery(this).addClass("active");
                var E = jQuery(this).attr("id");
                if (E == "current_background") {
                    settings.setBackgroundImage(_tmp.currentImage)
                } else {
                    if (E == "no_background") {
                        settings.setBackgroundImage("none")
                    }
                }
                return false
            });
            jQuery("#user_profile_background_tile").click(function() {
                var E = jQuery("#user_profile_background_tile:checked").val();
                jQuery("body").css({
                    "background-repeat": E ? "repeat": "no-repeat",
                    "background-attachment": E ? "scroll": "fixed"
                });
                jQuery("#user_profile_use_background_image").val("true")
            });
            settings.colorPicker = jQuery.farbtastic("#picker", settings.synchColors);
            jQuery("#picker").click(function() {
                jQuery(".colorwell-selected").focus()
            });
            var D;
            var C = (function() {
                var E = [8, 9, 13, 16, 17, 18, 37, 38, 39, 40, 45, 46, 86, 88, 90];
                for (var F = 48; F <= 57; F++) {
                    E.push(F)
                }
                for (var F = 65; F <= 70; F++) {
                    E.push(F)
                }
                for (var F = 96; F <= 105; F++) {
                    E.push(F)
                }
                return E
            })();
            jQuery(".colorwell").each(settings.synchColors).focus(function() {
                if (D) {
                    jQuery(D).removeClass("colorwell-selected")
                }
                settings.colorPicker.linked = this;
                settings.colorPicker.setColor(this.value);
                jQuery("#picker").show();
                jQuery(D = this).addClass("colorwell-selected")
            }).blur(function() {
                var E = jQuery(E);
                E.val(E.val().toUpperCase())
            }).keydown(function(E) {
                if (jQuery.inArray(E.keyCode, C) != -1) {
                    if (jQuery.inArray(E.keyCode, [86, 88, 90]) != -1) {
                        return (E.ctrlKey || E.metaKey)
                    } else {
                        if ((jQuery.inArray(E.keyCode, [8, 46]) != -1) || ((E.keyCode == 88) && E.shiftKey)) {
                            return (jQuery(this).val() != "#")
                        } else {
                            if (E.shiftKey) {
                                return (jQuery.inArray(E.keyCode, [65, 66, 67, 68, 69, 70]) != -1)
                            }
                        }
                    }
                    return true
                }
                return false
            }).keyup(function(E) {
                if (jQuery(this).val().indexOf("#") != 0) {
                    jQuery(this).val("#" + jQuery(this).val())
                }
                settings.colorPicker.setColor(this.value);
                var F = jQuery(this).val();
                if (!document.all || (F.length == 7 || F.length == 4)) {
                    jQuery(this).css({
                        "background-color": jQuery(this).val(),
                        color: settings.colorPicker.hsl[2] > 0.5 ? "#000000": "#ffffff"
                    });
                    settings.updateColors()
                }
            });
            jQuery("#design_customization form a").click(settings.updateColors)
        }
    };
    return A
} ();

jQuery(document).ready(function() {
		if($('settingsBox')) 
    	settings.initDesignSettings();
});
