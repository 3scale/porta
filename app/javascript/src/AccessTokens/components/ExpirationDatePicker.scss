@import '~@patternfly/patternfly/patternfly-addons.css';

.pf-c-calendar-month,
.pf-c-form-control-expiration {
  width: 50%;
}

.pf-c-form-control-expiration {
  margin-right: 1em;
}

button.pf-c-form__group-label-help {
  min-width: auto;
  width: auto;
}
