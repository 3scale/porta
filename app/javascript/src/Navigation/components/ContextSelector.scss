// Adjust styles for a context selector that is within the (dark) header
.header-context-selector {
  --pf-c-context-selector__toggle--BorderTopColor: transparent;
  --pf-c-context-selector__toggle--BorderRightColor: transparent;
  --pf-c-context-selector__toggle--BorderBottomColor: transparent;
  --pf-c-context-selector__toggle--BorderLeftColor: transparent;
  --pf-c-context-selector__toggle--active--BorderBottomColor: transparent;
  --pf-c-context-selector__toggle--expanded--BorderBottomColor: transparent;
  --pf-c-context-selector__toggle--hover--BorderBottomColor: transparent;
  --pf-c-context-selector--Width: initial;
}

.header-context-selector__toggle-text-icon {
  margin-right: var(--pf-global--spacer--sm);
  color: var(--pf-global--active-color--100);
}

.pf-c-context-selector__menu-list {
  overflow-y: auto;
}
