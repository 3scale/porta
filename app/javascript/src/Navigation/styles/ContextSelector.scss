@import '~@patternfly/patternfly/components/ContextSelector/context-selector.css';

// Adjust styles for a context selector that is within the (dark) header
.header-context-selector {
  --pf-c-context-selector__toggle--BorderTopColor: transparent;
  --pf-c-context-selector__toggle--BorderRightColor: transparent;
  --pf-c-context-selector__toggle--BorderBottomColor: transparent;
  --pf-c-context-selector__toggle--BorderLeftColor: transparent;
  --pf-c-context-selector__toggle--active--BorderBottomColor: transparent;
  --pf-c-context-selector__toggle--expanded--BorderBottomColor: transparent;
  --pf-c-context-selector__toggle--hover--BorderBottomColor: transparent;
  --pf-c-context-selector--Width: initial;
}

.header-context-selector__item-icon {
  // color: var(--pf-global--active-color--100);
  margin-right: var(--pf-global--spacer--sm);
  min-width: var(--pf-global--spacer--lg);
}

.header-context-selector__toggle-text-icon {
  margin-right: var(--pf-global--spacer--sm);
  color: var(--pf-global--active-color--100);
}


// // Copied from colors.scss
// $lochMaraBlue: #0088ce; //pf-blue-400
// $font-color: #393f44; // pf-black-800

// // scss-lint:disable SelectorFormat IdSelector
// #api_selector {
//   margin-bottom: 0.5rem; // var(--pf-c-page__header-tools--MarginTop)
//   margin-top: 0.5rem;
// }

// .current-context {
//   color: $lochMaraBlue;
// }
// .PopNavigation-list {
//   animation: none;
// }

// .PopNavigation-link > i {
//   padding-right: 0.5rem;
// }

// .unauthorized {
//   cursor: not-allowed;

//   &:hover {
//     color: $font-color;
//   }
// }
