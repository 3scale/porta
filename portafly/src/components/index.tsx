export * from './pages'
export * from './shared'
export * from './AppLayout'
export * from './AppNavExpandable'
export * from './AppNavGroup'
export * from './AppNavItem'
export * from './AppHeaderTools'
export * from './LazyRoute'
export * from './NotFound'
export * from './SwitchWith404'
export * from './util'
export * from './Root'
