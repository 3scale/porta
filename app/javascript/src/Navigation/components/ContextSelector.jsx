// @flow

import React from 'react'
import { ActiveMenuTitle } from 'Navigation/components/ActiveMenuTitle'
import { createReactWrapper } from 'utilities/createReactWrapper'

import 'Navigation/styles/ContextSelector.scss'

import type { Api, Service, Menu } from 'Types'

type Props = {
  apis: Api[],
  currentApi: Api,
  activeMenu: Menu,
  audienceLink: string
}

type State = {
  filterQuery: string
}

const DASHBOARD_PATH = '/p/admin/dashboard'

class ContextSelector extends React.Component<Props, State> {
  state = {
    filterQuery: ''
  }

  onFilterChange (event: SyntheticInputEvent<HTMLInputElement>) {
    const filterQuery = event.target.value.toLowerCase()
    this.setState({ filterQuery })
  }

  renderInput () {
    const { apis = [] } = this.props

    if (apis.length < 2) {
      return null
    }

    return (
      <li className="PopNavigation-listItem nav-search-widget docs-search" id="context-widget">
        <input
          onChange={(e: SyntheticInputEvent<HTMLInputElement>) => this.onFilterChange(e)}
          type="search"
          className="docs-search-input"
          placeholder="Type the API name"
        />
      </li>
    )
  }

  getClassNamesForMenu (menu: Menu): string {
    const { activeMenu } = this.props

    const isDashboardSelected = menu === 'dashboard' && activeMenu === 'dashboard'
    const isAudienceSelected = menu === 'audience' && (['buyers', 'finance', 'cms', 'site'].indexOf(activeMenu) !== -1)

    if (isDashboardSelected || isAudienceSelected) {
      return 'PopNavigation-link current-context'
    }

    return 'PopNavigation-link'
  }

  getClassNamesForService (service: Service): string {
    const { activeMenu, currentApi } = this.props
    let classNames = 'PopNavigation-link'

    if (['serviceadmin', 'monitoring'].indexOf(activeMenu) !== -1 &&
      service.id === currentApi.service.id) {
      classNames += ' current-context'
    }

    if (!service.link) {
      classNames += ' unauthorized'
    }

    return classNames
  }

  renderOptions () {
    const { apis } = this.props
    const { filterQuery } = this.state
    const filteredApis = apis.filter(api => api.service.name.toLowerCase().indexOf(filterQuery) !== -1)

    if (filteredApis.length === 0) {
      return null
    }

    const displayedApis = filteredApis.map(({ service }) => (
      <li key={service.id} className="PopNavigation-listItem">
        <a className={this.getClassNamesForService(service)} href={service.link}>
          <i className="fa fa-puzzle-piece" />{service.name}
        </a>
      </li>
    ))

    return (
      <li className="PopNavigation-listItem">
        <ul className="PopNavigation-results">
          {displayedApis}
        </ul>
      </li>
    )
  }

  render () {
    const { currentApi, activeMenu, audienceLink } = this.props

    return (
      <div className="PopNavigation PopNavigation--context">
        <a className="PopNavigation-trigger u-toggler" href="#context-menu" title="Context Selector">
          <ActiveMenuTitle currentApi={currentApi} activeMenu={activeMenu} />
        </a>
        <ul id="context-menu" className="PopNavigation-list u-toggleable">
          <li className="PopNavigation-listItem">
            <a className={this.getClassNamesForMenu('dashboard')} href={DASHBOARD_PATH}>
              <i className='fa fa-home' />Dashboard
            </a>
          </li>
          {audienceLink ? (
            <li className="PopNavigation-listItem">
              <a className={this.getClassNamesForMenu('audience')} href={audienceLink}>
                <i className='fa fa-bullseye' />Audience
              </a>
            </li>
          ) : null}
          {this.renderInput()}
          {this.renderOptions()}
        </ul>
      </div >
    )
  }
}

const ContextSelectorWrapper = (props: Props, containerId: string) => createReactWrapper(<ContextSelector {...props} />, containerId)

export { ContextSelector, ContextSelectorWrapper }
