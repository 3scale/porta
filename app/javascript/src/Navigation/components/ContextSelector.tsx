import { useRef, useState } from 'react'

import { ActiveMenuTitle } from 'Navigation/components/ActiveMenuTitle'
import { createReactWrapper } from 'utilities/createReactWrapper'
import { useClickOutside } from 'utilities/useClickOutside'

import type { FunctionComponent } from 'react'
import type { Menu } from 'Types'

import './ContextSelector.scss'

interface Props {
  activeMenu: Menu;
  audienceLink?: string;
  settingsLink: string;
  productsLink: string;
  backendsLink: string;
}

const DASHBOARD_PATH = '/p/admin/dashboard'

const ContextSelector: FunctionComponent<Props> = ({
  activeMenu,
  audienceLink,
  settingsLink,
  productsLink,
  backendsLink
}) => {
  const [isOpen, setIsOpen] = useState(false)
  const ref = useRef(null)
  useClickOutside(ref, () => { setIsOpen(false) })

  function getClassNamesForMenu (menu: Menu): string {
    const isDashboardSelected = menu === 'dashboard' && activeMenu === 'dashboard'
    const isAudienceSelected = menu === 'audience' && (['buyers', 'finance', 'cms', 'site'].includes(activeMenu))
    const isProductsSelected = menu === 'products' && (['serviceadmin', 'monitoring', 'products'].includes(activeMenu))
    const isBackendsSelected = menu === 'backend_api' && (['backend_api', 'backend_apis'].includes(activeMenu))
    const isSettingsSelected = menu === 'account' && (['account', 'personal', 'active_docs'].includes(activeMenu))

    if (isDashboardSelected || isAudienceSelected || isProductsSelected || isBackendsSelected || isSettingsSelected) {
      return 'pf-c-context-selector__menu-list-item current-context'
    }

    return 'pf-c-context-selector__menu-list-item'
  }

  return (
    <div className={`pf-c-context-selector header-context-selector ${isOpen ? ' pf-m-expanded' : ''}`} data-quickstart-id="context-selector" ref={ref}>
      <a className="pf-c-context-selector__toggle pf-m-plain pf-m-text" title="Context Selector" onClick={() => { setIsOpen(!isOpen) }}>
        <ActiveMenuTitle activeMenu={activeMenu} />
      </a>
      {isOpen && (
        <div className="pf-c-context-selector__menu">
          <ul className="pf-c-context-selector__menu-list" id="context-menu">
            <li>
              <a className={getClassNamesForMenu('dashboard')} href={DASHBOARD_PATH}>
                <i className="fa fa-home header-context-selector__item-icon" />Dashboard
              </a>
            </li>
            {!!audienceLink && (
              <li>
                <a className={getClassNamesForMenu('audience')} href={audienceLink}>
                  <i className="fa fa-bullseye header-context-selector__item-icon" />Audience
                </a>
              </li>
            )}
            <li>
              <a className={getClassNamesForMenu('products')} href={productsLink}>
                <i className="fa fa-cubes header-context-selector__item-icon" />Products
              </a>
            </li>
            <li>
              <a className={getClassNamesForMenu('backend_api')} href={backendsLink}>
                <i className="fa fa-cube header-context-selector__item-icon" />Backends
              </a>
            </li>
            <li>
              <a className={getClassNamesForMenu('account')} href={settingsLink}>
                <i className="fa fa-cog header-context-selector__item-icon" />Account Settings
              </a>
            </li>
          </ul>
        </div>
      )}
    </div >
  )
}

// eslint-disable-next-line react/jsx-props-no-spreading
const ContextSelectorWrapper = (props: Props, containerId: string): void => { createReactWrapper(<ContextSelector {...props} />, containerId) }

export type { Props }
export { ContextSelector, ContextSelectorWrapper }
