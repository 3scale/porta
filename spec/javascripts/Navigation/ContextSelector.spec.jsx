import React from 'react'
import { mount } from 'enzyme'
import { ContextSelector } from 'Navigation/components/ContextSelector'

const audienceLink = '/audience'
const productsLink = '/products'
const backendsLink = '/backends'

const currentApi = { name: 'api 0', id: 0, link: 'foo.bar' }

function getWrapper (customProps) {
  const props = {
    currentApi: null,
    activeMenu: '',
    audienceLink: audienceLink,
    productsLink: productsLink,
    backendsLink: backendsLink,
    ...customProps
  }
  return mount(<ContextSelector {...props} />)
}

it('should render itself', () => {
  expect(getWrapper().find(ContextSelector).exists()).toEqual(true)
})

it('should have Dashboard, Audience, Products, Backends and Settings', () => {
  const wrapper = getWrapper()
  wrapper.find('.pf-c-context-selector__toggle').simulate('click')
  expect(wrapper).toMatchSnapshot()
})

it('should not have Audience if not provided', () => {
  const wrapper = getWrapper()
  wrapper.setProps({ audienceLink: undefined })
  wrapper.find('.pf-c-context-selector__toggle').simulate('click')
  expect(wrapper).toMatchSnapshot()
})

it('should highlight the selected context', () => {
  const wrapper = getWrapper()
  wrapper.find('.pf-c-context-selector__toggle').simulate('click')

  wrapper.setProps({ activeMenu: 'buyers', currentApi: null })
  expect(wrapper.find('.current-context')).toHaveLength(1)
  expect(wrapper.find('.current-context').text()).toEqual('Audience')

  wrapper.setProps({ activeMenu: 'finance', currentApi: null })
  expect(wrapper.find('.current-context')).toHaveLength(1)
  expect(wrapper.find('.current-context').text()).toEqual('Audience')

  wrapper.setProps({ activeMenu: 'cms', currentApi: null })
  expect(wrapper.find('.current-context')).toHaveLength(1)
  expect(wrapper.find('.current-context').text()).toEqual('Audience')

  wrapper.setProps({ activeMenu: 'site', currentApi: null })
  expect(wrapper.find('.current-context')).toHaveLength(1)
  expect(wrapper.find('.current-context').text()).toEqual('Audience')

  wrapper.setProps({ activeMenu: 'dashboard', currentApi })
  expect(wrapper.find('.current-context')).toHaveLength(1)
  expect(wrapper.find('.current-context').text()).toEqual('Dashboard')

  wrapper.setProps({ activeMenu: 'personal', currentApi })
  expect(wrapper.find('.current-context')).toHaveLength(1)
  expect(wrapper.find('.current-context').text()).toEqual('Account Settings')

  wrapper.setProps({ activeMenu: 'account', currentApi })
  expect(wrapper.find('.current-context')).toHaveLength(1)
  expect(wrapper.find('.current-context').text()).toEqual('Account Settings')

  wrapper.setProps({ activeMenu: 'active_docs', currentApi })
  expect(wrapper.find('.current-context')).toHaveLength(1)
  expect(wrapper.find('.current-context').text()).toEqual('Account Settings')

  wrapper.setProps({ activeMenu: 'serviceadmin', currentApi })
  expect(wrapper.find('.current-context')).toHaveLength(1)
  expect(wrapper.find('.current-context').text()).toEqual('Products')

  wrapper.setProps({ activeMenu: 'backend_api', currentApi })
  expect(wrapper.find('.current-context')).toHaveLength(1)
  expect(wrapper.find('.current-context').text()).toEqual('Backends')
})

it('should display the current api', () => {
  const wrapper = getWrapper({ activeMenu: 'serviceadmin', currentApi })

  expect(wrapper.find('.fa-cubes').exists()).toBe(true)
  expect(wrapper.find('.pf-c-context-selector__toggle').text()).toContain('Products')

  wrapper.setProps({ activeMenu: 'monitoring' })
  expect(wrapper.find('.fa-cubes').exists()).toBe(true)
  expect(wrapper.find('.pf-c-context-selector__toggle').text()).toContain('Products')

  wrapper.setProps({ activeMenu: 'backend_api' })
  expect(wrapper.find('.fa-cube').exists()).toBe(true)
  expect(wrapper.find('.pf-c-context-selector__toggle').text()).toContain('Backends')
})
