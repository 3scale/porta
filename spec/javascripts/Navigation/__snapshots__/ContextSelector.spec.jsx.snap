// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should have Dashboard, Audience, Products, Backends and Settings 1`] = `
<ContextSelector
  activeMenu=""
  audienceLink="/audience"
  backendsLink="/backends"
  currentApi={null}
  productsLink="/products"
>
  <div
    className="pf-c-context-selector header-context-selector  pf-m-expanded"
  >
    <a
      className="pf-c-context-selector__toggle "
      href="#context-menu"
      onClick={[Function]}
      title="Context Selector"
    >
      <ActiveMenuTitle
        activeMenu=""
      >
        <span
          className="pf-c-context-selector__toggle-text"
        >
          <i
            className="fa  header-context-selector__toggle-text-icon"
          />
        </span>
        <i
          className="fa fa-chevron-down  pf-c-context-selector__toggle-icon"
        />
      </ActiveMenuTitle>
    </a>
    <div
      className="pf-c-context-selector__menu"
    >
      <ul
        className="pf-c-context-selector__menu-list"
        id="context-menu"
      >
        <li>
          <a
            className="pf-c-context-selector__menu-list-item"
            href="/p/admin/dashboard"
          >
            <i
              className="fa fa-home header-context-selector__item-icon"
            />
            Dashboard
          </a>
        </li>
        <li>
          <a
            className="pf-c-context-selector__menu-list-item"
            href="/audience"
          >
            <i
              className="fa fa-bullseye header-context-selector__item-icon"
            />
            Audience
          </a>
        </li>
        <li>
          <a
            className="pf-c-context-selector__menu-list-item"
            href="/products"
          >
            <i
              className="fa fa-cubes header-context-selector__item-icon"
            />
            Products
          </a>
        </li>
        <li>
          <a
            className="pf-c-context-selector__menu-list-item"
            href="/backends"
          >
            <i
              className="fa fa-cube header-context-selector__item-icon"
            />
            Backends
          </a>
        </li>
        <li>
          <a
            className="pf-c-context-selector__menu-list-item"
          >
            <i
              className="fa fa-cog header-context-selector__item-icon"
            />
            Account Settings
          </a>
        </li>
      </ul>
    </div>
  </div>
</ContextSelector>
`;

exports[`should not have Audience if not provided 1`] = `
<ContextSelector
  activeMenu=""
  backendsLink="/backends"
  currentApi={null}
  productsLink="/products"
>
  <div
    className="pf-c-context-selector header-context-selector  pf-m-expanded"
  >
    <a
      className="pf-c-context-selector__toggle "
      href="#context-menu"
      onClick={[Function]}
      title="Context Selector"
    >
      <ActiveMenuTitle
        activeMenu=""
      >
        <span
          className="pf-c-context-selector__toggle-text"
        >
          <i
            className="fa  header-context-selector__toggle-text-icon"
          />
        </span>
        <i
          className="fa fa-chevron-down  pf-c-context-selector__toggle-icon"
        />
      </ActiveMenuTitle>
    </a>
    <div
      className="pf-c-context-selector__menu"
    >
      <ul
        className="pf-c-context-selector__menu-list"
        id="context-menu"
      >
        <li>
          <a
            className="pf-c-context-selector__menu-list-item"
            href="/p/admin/dashboard"
          >
            <i
              className="fa fa-home header-context-selector__item-icon"
            />
            Dashboard
          </a>
        </li>
        <li>
          <a
            className="pf-c-context-selector__menu-list-item"
            href="/products"
          >
            <i
              className="fa fa-cubes header-context-selector__item-icon"
            />
            Products
          </a>
        </li>
        <li>
          <a
            className="pf-c-context-selector__menu-list-item"
            href="/backends"
          >
            <i
              className="fa fa-cube header-context-selector__item-icon"
            />
            Backends
          </a>
        </li>
        <li>
          <a
            className="pf-c-context-selector__menu-list-item"
          >
            <i
              className="fa fa-cog header-context-selector__item-icon"
            />
            Account Settings
          </a>
        </li>
      </ul>
    </div>
  </div>
</ContextSelector>
`;
